docs/source/getting_started/benchmarks/imagenette_benchmark.py

```diff
@@ -0,0 +1,441 @@
+        last_conv_channels = list(resnet.children())[-1].in_features
+        self.backbone = nn.Sequential(
+            *list(resnet.children())[:-1],
+            nn.Conv2d(last_conv_channels, num_ftrs, 1),
+        )
+        # create a byol model based on ResNet
+        self.resnet_byol = \
+            lightly.models.BYOL(self.backbone, num_ftrs=num_ftrs)
+        self.criterion = lightly.loss.SymNegCosineSimilarityLoss()
+
+    def forward(self, x):
+        self.resnet_byol(x)
+
+    def training_step(self, batch, batch_idx):
+        (x0, x1), _, _ = batch
+        x0, x1 = self.resnet_byol(x0, x1)
+        loss = self.criterion(x0, x1)
+        self.log('train_loss_ssl', loss)
+        return loss
+
+    def configure_optimizers(self):
+        optim = torch.optim.SGD(self.resnet_byol.parameters(), lr=6e-2,
+                                momentum=0.9, weight_decay=5e-4)
+        scheduler = torch.optim.lr_scheduler.CosineAnnealingLR(optim, max_epochs)
+        return [optim], [scheduler]
+
+class NNBYOLModel(BenchmarkModule):
+    def __init__(self, dataloader_kNN, num_classes):
+        super().__init__(dataloader_kNN, num_classes)
+        # create a ResNet backbone and remove the classification head
+        resnet = torchvision.models.resnet18()
+        last_conv_channels = list(resnet.children())[-1].in_features
+        self.backbone = nn.Sequential(
+            *list(resnet.children())[:-1],
+            nn.Conv2d(last_conv_channels, num_ftrs, 1),
+        )
+        # create a byol model based on ResNet
+        self.resnet_byol = \
+            lightly.models.BYOL(self.backbone, num_ftrs=num_ftrs)
+        self.criterion = lightly.loss.SymNegCosineSimilarityLoss()
+
+        self.nn_replacer = NNMemoryBankModule(size=nn_size)
+
+    def forward(self, x):
+        self.resnet_byol(x)
+
+    def training_step(self, batch, batch_idx):
+        (x0, x1), _, _ = batch
+        (z0, p0), (z1, p1) = self.resnet_byol(x0, x1)
+        z0 = self.nn_replacer(z0.detach(), update=False)
+        z1 = self.nn_replacer(z1.detach(), update=True)
+        loss = self.criterion((z0, p0), (z1, p1))
+        self.log('train_loss_ssl', loss)
+        return loss
+
+    def configure_optimizers(self):
+        optim = torch.optim.SGD(self.resnet_byol.parameters(), lr=6e-2,
+                                momentum=0.9, weight_decay=5e-4)
+        scheduler = torch.optim.lr_scheduler.CosineAnnealingLR(optim, max_epochs)
+        return [optim], [scheduler]
+
+model_names = ['MoCo_256', 'SimCLR_256', 'SimSiam_256', 'BarlowTwins_256', 
+               'BYOL_256', 'NNCLR_256', 'NNSimSiam_256', 'NNBYOL_256']
+
+models = [MocoModel, SimCLRModel, SimSiamModel, BarlowTwinsModel, 
+          BYOLModel, NNCLRModel, NNSimSiamModel, NNBYOLModel]
+
+bench_results = []
+gpu_memory_usage = []
+
+# loop through configurations and train models
+for batch_size in batch_sizes:
+    for model_name, BenchmarkModel in zip(model_names, models):
```

Comment:
"BenchmarkModel" is a class and thus unsuited as variable name, this overwrites the class.